crates/oxc_linter/src/rules/eslint/yoda.rs

```diff
@@ -0,0 +1,1160 @@
+use oxc_ast::{
+    ast::{
+        BinaryExpression, BinaryOperator, Expression, LogicalExpression, LogicalOperator,
+        UnaryOperator,
+    },
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_ecmascript::ToBigInt;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+use regex::Regex;
+
+use crate::{context::LintContext, rule::Rule, utils::is_same_reference, AstNode};
+
+fn yoda_diagnostic(span: Span) -> OxcDiagnostic {
```

Comment:
shall we have a few diagnostics based on the config struct?
doing thsi should help make the help msg easier to comprehend

Reply:
Thanks!
I fixed refer to [original rule](https://github.com/eslint/eslint/blob/e6b84f535e84050b0c63ae304eb17f9d181ac299/lib/rules/yoda.js#L146) 
https://github.com/oxc-project/oxc/pull/7559/commits/e07f9fdf7f014a655128c82f7fdbfd53c762978f